waltid-services/waltid-wallet-api/src/main/kotlin/id/walt/webwallet/web/controllers/AuthController.kt

```diff
@@ -295,15 +295,30 @@ fun Application.auth() {
                     response {
                         HttpStatusCode.Created to { description = "Registration succeeded " }
                         HttpStatusCode.BadRequest to { description = "Registration failed" }
+                        HttpStatusCode.Conflict to { description = "Account already exists!" }
                     }
                 }) {
-                val req = loginRequestJson.decodeFromString<AccountRequest>(call.receive())
-                AccountsService.register("", req)
+                val req = call.receiveText()
+                val jsonElement = Json.parseToJsonElement(req)
+                val jsonObject = jsonElement.jsonObject
+
+                if (!jsonObject.containsKey("type") || jsonObject["type"]?.jsonPrimitive?.content.isNullOrEmpty()) {
+                    call.respond(HttpStatusCode.BadRequest, "Missing type in request")
+                    return@post
+                }
+                val accountRequest = loginRequestJson.decodeFromString<AccountRequest>(call.receive())
+                AccountsService.register("", accountRequest)
                     .onSuccess {
                         call.response.status(HttpStatusCode.Created)
                         call.respond("Registration succeeded ")
                     }
-                    .onFailure { call.respond(HttpStatusCode.BadRequest, it.localizedMessage) }
+                    .onFailure {
+                        if (it.message?.contains("Account already exists!") == true) {
```

Comment:
comparing against the error message to compute the error status doesn't feel right
maybe we can have a dedicated type for application logic errors
e.g. have an entity representing an `Error`:
```kotlin
abstract class Error(
    val code: String,
    override val message: String,
) : Throwable()
```
and then define for each module (Account, Did, Key, etc.) such errors, e.g.:
```kotlin
object Errors {
    object General {
        class Conflict(type: KClass<*>, data: String) = Error(
            code = "conflict.already.exists",
            message = "${type.getFullName()} for $data already exists."
        )
    }
}
```
then throw an instance of this error (e.g. `throw Errors.General.Conflict(Account::class, request.email)`)

Reply:
computing the status could be done potentially using ktor StatusPages:
```kotlin
install(StatusPages) {
    exception<Throwable> { call, cause ->
        if(cause is Errors.General.Conflict) {
            call.respondText(text = "409: $cause" , status = HttpStatusCode.Conflict)
        } else {
            call.respondText(text = "500: $cause" , status = HttpStatusCode.InternalServerError)
        }
    }
}
```

Reply:
or extending the Error class with a method that inheritors would have to implement
e.g.
```kotlin
abstract class Error(
    val code: String,
    override val message: String,
) : Throwable() {
	abstract fun toHttpStatus(): HttpStatusCode
}

object Errors {
    object General {
        class Conflict(type: KClass<*>, data: String) = Error(
            code = "conflict.already.exists",
            message = "${type.getFullName()} for $data already exists."
        ){
        	override fun toHttpStatus(): HttpStatusCode = HttpStatusCode.Conflict
        }
    }
}
```

Reply:
will implement this feature in a separate branch